src/runtime/pkg/katautils/katatrace/tracing.go

```diff
@@ -163,8 +164,76 @@ func Trace(parent context.Context, logger *logrus.Entry, name string, tags ...ma
 	return span, ctx
 }
 
-// AddTag adds an additional key-value pair to a tracing span. This can be used to
-// provide dynamic tags that are determined at runtime.
-func AddTag(span otelTrace.Span, key string, value interface{}) {
-	span.SetAttributes(label.Any(key, value))
+func addTag(span otelTrace.Span, key string, value interface{}) {
+	// do not append tags if tracing is disabled
+	if tracing {
```

Comment:
Nit: Just a personal preference, but I'd invert this to remove one level of indent:

```go
if !tracing {
  return
}

// ...
```

Reply:
@jodh-intel updated